test/conformance/helpers/broker_tracing_test_helper.go

```diff
@@ -0,0 +1,489 @@
+	}
+
+	for n, tc := range testCases {
+		loggerPodName := "logger"
+		t.Run(n, func(t *testing.T) {
+			channelTestRunner.RunTests(t, common.FeatureBasic, func(st *testing.T, channel string) {
+				// Don't accidentally use t, use st instead. To ensure this, shadow 't' to a useless
+				// type.
+				t := struct{}{}
+				_ = fmt.Sprintf("%s", t)
+
+				client := common.Setup(st, true)
+				defer common.TearDown(client)
+
+				// Label namespace so that it creates the default broker.
+				if err := client.LabelNamespace(map[string]string{"knative-eventing-injection": "enabled"}); err != nil {
+					st.Fatalf("Error annotating namespace: %v", err)
+				}
+
+				// Do NOT call zipkin.CleanupZipkinTracingSetup. That will be called exactly once in
+				// TestMain.
+				tracinghelper.Setup(st, client)
+
+				expected, mustContain := setupBrokerTracing(st, channel, client, loggerPodName, tc.incomingTraceId)
+				assertLogContents(st, client, loggerPodName, mustContain)
+				traceID := getTraceID(st, client, loggerPodName)
+				trace, err := zipkin.JSONTrace(traceID, expected.SpanCount(), 2*time.Minute)
+				if err != nil {
+					st.Fatalf("Unable to get trace %q: %v. Trace so far %+v", traceID, err, tracinghelper.PrettyPrintTrace(trace))
+				}
+
+				tree := tracinghelper.GetTraceTree(st, trace)
+				if err := expected.Matches(tree); err != nil {
+					st.Fatalf("Trace Tree did not match expected: %v", err)
+				}
+			})
+		})
+	}
+}
+
+// setupBrokerTracing is the general setup for TestBrokerTracing. It creates the following:
+// 1. Broker.
+// 2. Trigger on 'foo' events -> K8s Service -> transformer Pod (which replies with a 'bar' event).
+// 3. Trigger on 'bar' events -> K8s Service -> eventdetails Pod.
+// 4. Sender Pod which sends a 'foo' event.
+// It returns a string that is expected to be sent by the SendEvents Pod and should be present in
+// the LogEvents Pod logs.
+func setupBrokerTracing(t *testing.T, channel string, client *common.Client, loggerPodName string, incomingTraceId bool) (tracinghelper.TestSpanTree, string) {
+	// Create the Broker.
+	const (
+		brokerName    = "br"
+		etTransformer = "transformer"
+		etLogger      = "logger"
+	)
+	channelTypeMeta := common.GetChannelTypeMeta(channel)
+	client.CreateBrokerOrFail(brokerName, channelTypeMeta)
+
+	// Due to https://github.com/knative/eventing/issues/1998, wait for the Broker to become ready
+	// before creating the Triggers.
+	err := client.WaitForResourceReady(brokerName, &metav1.TypeMeta{
+		APIVersion: resources.EventingAPIVersion,
+		Kind:       "Broker",
+	})
+	if err != nil {
+		t.Fatalf("Broker did not become ready: %v", err)
+	}
+
+	// Create a logger (EventDetails) Pod and a K8s Service that points to it.
+	logPod := resources.EventDetailsPod(loggerPodName)
+	client.CreatePodOrFail(logPod, common.WithService(loggerPodName))
+
+	// Create a Trigger that receives events (type=bar) and sends them to the logger Pod.
+	loggerTrigger := client.CreateTriggerOrFail(
+		"logger",
```

Comment:
Define it as a constant?

Reply:
I could, but why? This is the only place the value is used. Everywhere else it is needed, we use `loggerTrigger.Name`.